docs2/site/docs/getting-started/schema-types.md

```diff
@@ -112,21 +163,92 @@ public class EpisodeEnum : EnumerationGraphType
+
+**Resolving Enumerations**
+
+Fields typed as enumerations are resolved by returning either the name or backing value of one or more of the enum members. In the below examples, notice the identical implementations of the `appearsIn` field for both human graph types. In both implementations, the client receives the enum member names in response to queries on the `appearsIn` field.
+
+If the field resolves a value which cannot be mapped to one of the enum's legal values, GraphQL.NET will return `null` to the client in the data for the field.
 
 ```csharp
-public class EpisodeEnum : EnumerationGraphType<Episodes>
+public class HumanString
 {
+    //i.e. "NEWHOPE", "EMPIRE", "JEDI"
+    public string[] AppearsIn { get; set; }
 }
-```
 
-**.NET**
+public class HumanStringType: ObjectGraphType<HumanString>
+{
+    public HumanStringType()
+    {
+        Name = "HumanString";
+        Field<ListGraphType<EpisodeEnum>>("appearsIn", "Which movie they appear in.");
+    }
+}
 
-```csharp
-public enum Episodes
+public class HumanInt
 {
-    NEWHOPE  = 4,
-    EMPIRE  = 5,
-    JEDI  = 6
+    //i.e. 4, 5, 6
+    public int AppearsIn { get; set; }
 }
+
+public class HumanIntType: ObjectGraphType<HumanInt>
+{
+    public HumanIntType()
+    {
+        Name = "HumanInt";
+        Field<ListGraphType<EpisodeEnum>>("appearsIn", "Which movie they appear in.");
+    }
+}
+```
+
+**Enumeration Arguments**
+
+Enumerations can be used as arguments in GraphQL queries. Consider a query which gets the humans appearing in a specific film:
+
+```graphql
+query HumansAppearingIn($episode: Episode){
+    humans(appearsIn: $episode){
+        id
+        name
+        appearsIn
+    }
+}
+
+# example query variables:
+# {
+#   "episode":"NEWHOPE"
+# }
+```
+
+When GraphQL.NET receives an enum member name as a query argument, the queried field's `ResolveFieldContext` stores the backing value associated with the enum member name in its arguments list. The GraphQL.NET query type which handles the example query may be implemented as:
+
+```csharp
+    public class StarWarsQuery : ObjectGraphType<object>
+    {
+        public StarWarsQuery()
+        {
+            Name = "Query";
+
+            Field<ListGraphType<HumanType>>(
+                "humans",
+                arguments: new QueryArguments(
+                    new QueryArgument<EpisodeEnum> 
+                        { 
+                            Name = "appearsIn", 
+                            Description = "An episode the human appears in." 
+                        }
+                ),
+                resolve: context => 
+                {
+                    //episode = 4
+                    var episode = context.GetArgument<int>("appearsIn");
```

Comment:
It will be useful to show that `var episode = context.GetArgument<Episodes>("appearsIn");` works too.